internal/bminventory/inventory.go

```diff
@@ -340,9 +341,51 @@ func (b *bareMetalInventory) setDefaultRegisterClusterParams(ctx context.Context
 		}
 	}
 
+	params.NewClusterParams.NetworkType, err = getDefaultNetworkType(params)
+	if err != nil {
+		return params, err
+	}
+
 	return params, nil
 }
 
+// If the cluster is SNO, or the IP version is IPv6 (based on ClusterNetworkCidr param), or the OpenShift version >= 4.12,
+// the network type will be set to "OVNKubernetes", otherwise it will be set to "OpenShiftSDN"
+func getDefaultNetworkType(params installer.V2RegisterClusterParams) (*string, error) {
+	// If the network type is specified already, we return it
+	if params.NewClusterParams.NetworkType != nil {
+		return params.NewClusterParams.NetworkType, nil
+	}
+
+	isOpenShiftVersionRecentEnough, err := verifyMinimalOpenShiftVersionForDefaultNetworkTypeOVNKubernetes(swag.StringValue(params.NewClusterParams.OpenshiftVersion))
+	if err != nil {
+		return params.NewClusterParams.NetworkType, err
+	}
+
+	isSingleNodeCluster := swag.StringValue(params.NewClusterParams.HighAvailabilityMode) == models.ClusterCreateParamsHighAvailabilityModeNone
+
+	if isOpenShiftVersionRecentEnough || isSingleNodeCluster {
+		return swag.String(models.ClusterCreateParamsNetworkTypeOVNKubernetes), nil
+	} else {
+		return swag.String(models.ClusterCreateParamsNetworkTypeOpenShiftSDN), nil
+	}
+}
+
+func verifyMinimalOpenShiftVersionForDefaultNetworkTypeOVNKubernetes(requestedOpenshiftVersion string) (bool, error) {
+	ocpVersion, err := version.NewVersion(requestedOpenshiftVersion)
+	if err != nil {
+		return false, errors.Errorf("Failed to parse OCP version %s", requestedOpenshiftVersion)
+	}
+	minimalVersionForSno, err := version.NewVersion(minimalOpenShiftVersionForDefaultNetworkTypeOVNKubernetes)
```

Comment:
why did the member call minimalVersionForSno? it's not relevant to SNO
should be minimalVersionForNetworkTypeOVN or something similar

Reply:
Mistake, was corrected now